lib/buffer.js

```diff
@@ -370,8 +370,12 @@ Buffer.prototype.toString = function() {
   } else {
     var result = slowToString.apply(this, arguments);
   }
-  if (result === undefined)
-    throw new Error('toString failed');
+  if (result === undefined) {
+    var msg = 'toString() failed: buffer larger than maximum string size ';
+    msg += binding.kStringMaxLength;
+    msg += ' bytes';
+    throw new RangeError(msg);
+  }
```

Comment:
Yeah, I kinda like that style better too. Nice job looking into this further guys! :+1: